### ISSUE TITLE
Any nullable props crashed build

### ISSUE THREAD

**Issue by nestle49** [qmhc/vite-plugin-dts#187]
### Describe the bug

Build crashed if any props equals null

### Reproduction

minimal repo 

### Steps to reproduce

```
export default defineConfig({
    plugins: [
        vue(),
        dts({
            insertTypesEntry: true
        }),
        typescript2({
            check: false,
            tsconfigOverride: {
                compilerOptions: {
                    outDir: 'dist',
                    sourceMap: true,
                    declaration: true,
                    declarationMap: true
                }
            },
            exclude: ['vite.config.ts']
        })
    ],
    resolve: {
        alias: {
            '~': path.resolve(__dirname, 'src')
        }
    },
    build: {
        cssCodeSplit: true,
        lib: {
            // Could also be a dictionary or array of multiple entry points
            entry: resolve(__dirname, 'src/main.ts'),
            name: 'UiKit',
            // the proper extensions will be added
            fileName: 'ui-kit'
        },
        rollupOptions: {
            // make sure to externalize deps that shouldn't be bundled
            // into your library
            external: [
                'vue',
                '@vueuse/core',
                'vue-router',
                /src\/App.vue/ // exclude App.vue component
            ],
            output: {
                assetFileNames: (assetInfo) => {
                    if (assetInfo.name === 'main.css') {
                        return 'ui-kit.css';
                    }
                    return assetInfo.name || '';
                },
                exports: 'named',

                // Provide global variables to use in the UMD build
                // for externalized deps
                globals: {
                    vue: 'Vue'
                }
            }
        }
    }
});

```

And use any nullable props, for example 

```
interface Props {
    value?: File[] | null;

const props = withDefaults(defineProps<Props>(), {
    value: null
});
```

Next npm run build

```
src/components/UiFileInput.vue.ts:23:5 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Type '{ type: (ArrayConstructor | null)[]; required: false; default: null; }' is not assignable to type 'Prop<unknown, unknown> | null'.
      Types of property 'type' are incompatible.
        Type '(ArrayConstructor | null)[]' is not assignable to type 'true | PropType<unknown> | null | undefined'.
          Type '(ArrayConstructor | null)[]' is not assignable to type 'PropConstructor<unknown>[]'.
            Type 'ArrayConstructor | null' is not assignable to type 'PropConstructor<unknown>'.
              Type 'null' is not assignable to type 'PropConstructor<unknown>'.

23     value: { type: [Array, null], required: false, default: null },
       ~~~~~

  node_modules/@vue/runtime-core/dist/runtime-core.d.ts:666:25
    666 export declare function defineComponent<PropsOptions extends Readonly<ComponentPropsOptions>, RawBindings, D, C extends ComputedOptions = {}, M extends MethodOptions = {}, Mixin extends ComponentOptionsMixin = ComponentOptionsMixin, Extends extends ComponentOptionsMixin = ComponentOptionsMixin, E extends EmitsOptions = {}, EE extends string = string, I extends ComponentInjectOptions = {}, II extends string = string>(options: ComponentOptionsWithObjectProps<PropsOptions, RawBindings, D, C, M, Mixin, Extends, E, EE, I, II>): DefineComponent<PropsOptions, RawBindings, D, C, M, Mixin, Extends, E, EE>;
                                ~~~~~~~~~~~~~~~
    The last overload is declared here.
```

### System Info

```shell
System:
    OS: macOS 13.0.1
    CPU: (8) arm64 Apple M1
    Memory: 208.38 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
  Browsers:
    Chrome: 110.0.5481.177
    Firefox: 106.0.1
    Safari: 16.1
  npmPackages:
    @vitejs/plugin-vue: ^4.0.0 => 4.0.0 
    vite: ^4.1.1 => 4.1.4 
    vite-plugin-dts: ^1.7.2 => 1.7.2
```

### Validations

- [X] Read the [FAQ](https://github.com/qmhc/vite-plugin-dts#faq).
- [X] Check that there isn't [already an issue](https://github.com/qmhc/vite-plugin-dts/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by qmhc** [comment]
Would provide a reproduction?

**Comment by nestle49** [comment]
Sorry, maybe later

This error observed inside any ui-kit (vue component library who based on vite and dts)

I found this problem https://github.com/qmhc/vite-plugin-dts/issues/142

If I remove nullable value for props - all fine, but I don't understand why I can't use nullable props

**Comment by qmhc** [comment]
Fixed in `3.0.0`.
